### ISSUE TITLE
Formgroup with nested formgroup throw exception on null values

### ISSUE THREAD

**Issue by szanellato93** [angular/angular#20509]
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[X] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
When i call form reset on a formGroup with nested formGroup passing it with a null value i receive a error "Cannot read property ... of null"

## Expected behavior
Expected result is that on form reset with a null value the nested form group properties become all null. 

## Minimal reproduction of the problem with instructions
When i have a formGroup with a nested formGroup like this:

```
let form = this.fb.group({
Id: [null],
Nested: fb.group({
    Id: [null]
})
})
```
If i call a form reset or a patch value with a object like this:
```
form.reset({
Id: 1,
Nested: null
})
```
i receive the error "Cannot read property 'Id' of null" because he try to read the nested property from a null object

Expected result is that on form reset with a null value the nested form group properties become all null. This behavior happen when i call the reset without the nested properties:
`form.reset({Id: 3})`

## What is the motivation / use case for changing the behavior?
Usually service return models with nested models with the value null.

**Comment by artem-galas** [comment]
I would like to take this issue.
But I'm not sure how to implement it.

The error occurred on `.reset` method for FormGroup. https://github.com/angular/angular/blob/master/packages/forms/src/model.ts#L1478

We can check `if (control instanceof FormGroup)` into this method, but I'm not sure if am I right. 

Here is my idea: 
As I understand we have to iterate through all controls from nested `formGroup` and reset value to null.
So basically we can call `control._forEachChild`.

Could you please advise me a little here?

Thanks.

**Comment by timrasche** [comment]
Any update on this?

**Comment by w5l** [comment]
This still happens, currently caused by the `reset` code in `FormGroup` here:

https://github.com/angular/angular/blob/bd34bc9e89f18a4d3c4fc633bf5624bfbdc3d1a3/packages/forms/src/model.ts#L1677-L1684

It blindly assumes `value` is set. This seems like a simple fix, whats the proper way to do a `null` check in typescript/javascript?

**Comment by dylhunn** [comment]
We'll need to see how breaking this fix is. I really hope it's not too breaking in g3 -- that would mean people are relying on app crashed 🤦
